src/Raven.Server/Documents/Queries/QueryRunner.cs

```diff
@@ -92,7 +94,11 @@ public Task ExecutePatchQuery(string indexName, IndexQuery query, QueryOperation
             if (options.AllowStale == false && results.IsStale)
                 throw new InvalidOperationException("Cannot perform delete operation. Query is stale.");
 
-            foreach (var document in results.Results)
+            IEnumerable<Document> documents = results.Results;
+            if (options.MaxOpsPerSecond.HasValue)
+                documents = documents.LimitRate(options.MaxOpsPerSecond.Value, TimeSpan.FromSeconds(1));
+
+            foreach (var document in documents)
```

Comment:
We are batching this with BatchSize of 1024 (const). It can be problematic with MaxOpsPerScond that have a very low value. We can e.g. require that his value can be no less than 100? What do you think?